### PR TITLE
use-endpoint-instead-of-localhost

### PR DESCRIPTION
**Pull request by qiaozha**
# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/resiliency/srv-driven/main.tsp [hunk 1]
```diff
@@ -25,9 +25,12 @@ using TypeSpec.Http;
 })
 @service
 @server(
-  "http://localhost:3000/resiliency/service-driven/client:v2/service:{serviceDeploymentVersion}/api-version:{apiVersion}",
+  "{endpoint}/resiliency/service-driven/client:v2/service:{serviceDeploymentVersion}/api-version:{apiVersion}",
   "Testserver endpoint",
   {
+    @doc("Need to be set as 'http://localhost:3000' in client.")
+    endpoint: url,
```
Comment (qiaozha) [inline packages/cadl-ranch-specs/http/resiliency/srv-driven/main.tsp:32]: Should we set a default value for it ?

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/resiliency/srv-driven/main.tsp:32]: I don't think we has client-default in tsp?

**Review by weidongxu-microsoft — APPROVED** [review]
make sure your local test is good.

**Comment by qiaozha** [comment]
> make sure your local test is good.

Yeah, I have tested it in https://github.com/Azure/autorest.typescript/pull/2032, it should work.

**Review by tadelesh — APPROVED** [review]

**Review by chunyu3 — APPROVED** [review]

**Pull request merged**